src/metrics/print.rs

```diff
@@ -1,23 +1,95 @@
-use super::Consumer;
+use std::io::Write;
+use std::sync::{Arc, Mutex};
+
 use async_trait::async_trait;
 use tokio::sync::mpsc::Receiver;
 
-pub struct Print;
+use super::Consumer;
+
+pub struct Print<T: 'static + Write + Send> {
+    csv_writer: Arc<Mutex<csv::Writer<T>>>,
+}
+
+impl<T: 'static + Write + Send> Print<T> {
+    pub fn new(writer: T) -> Self {
+        Self {
+            csv_writer: Arc::new(Mutex::new(csv::Writer::from_writer(writer))),
+        }
+    }
+
+    async fn write_record_not_blocking(&self, record: Vec<String>) -> Result<(), String> {
+        let csv_writer_clone = self.csv_writer.clone();
+
+        tokio::task::spawn_blocking(move || {
+            csv_writer_clone
+                .lock()
+                .map_err(|error| format!("Failed to acquire lock with error: {}", error))
```

Comment:
I call several times `map_err` to convert the error to a string, maybe it is better to change the error returned by the consume to `anyhow::Error`

Reply:
`anyhow::Error` would definitely be preferred, I think

Reply:
thanks , it can be addressed in another PR since we already have an issue for it: #28